include/grpc/grpc_security.h

```diff
@@ -290,11 +290,22 @@ GRPCAPI grpc_channel *grpc_secure_channel_create(
 
 typedef struct grpc_server_credentials grpc_server_credentials;
 
+typedef grpc_get_server_credentials_result (
+    *grpc_get_server_credentials_callback)(grpc_server_credentials **creds,
+                                           void *cb_arg);
+
 /** Releases a server_credentials object.
    The creator of the server_credentials object is responsible for its release.
    */
 GRPCAPI void grpc_server_credentials_release(grpc_server_credentials *creds);
 
+/** Extended options for grpc_ssl_server_credentials_create. */
+typedef struct {
+  grpc_get_server_credentials_callback get_server_credentials_cb;
+  void *get_server_credentials_cb_arg;
+  void *reserved;
```

Comment:
I guess since the extra options are part of a struct, we don't need `reserved` anymore?

Reply:
Keeping the `reserved` gives a way to introduce another struct of options later on. I feel like this is a bit clunky, but safe. If we omitted `reserved` and just added to the struct, I have some concern that we may not be able to determine if the caller is providing the old or new struct format. If we can convince ourselves that this won't ever be an issue, then I'm okay with omitting.

Reply:
> If we omitted reserved and just added to the struct, I have some concern that we may not be able to determine if the caller is providing the old or new struct format.

Hmm are you referring to binary compatibility concern where user compiles his application against older `options` struct but at run time runs with newer library that has newer/different `options` struct? If so I'm not sure how `reserved` would help detect? Is it a standard/generally accepted solution to have the `reserved` field as a solution?

Reply:
discussed during video conference, that `reserved` can be omitted if we provide an opaque way to initialize/set options, e.g.,
```
options* new_[default_]options();
void options_set_cb(...);
void options_set_cb_arg(...);
```
etc.
@justinburke maybe an example is `gpr_thd_options` in `tools/distrib/python/grpcio_tools/grpc_root/include/grpc/support/thd.h`

Reply:
Thanks for reference. Done.